### PR TITLE
sygusComp 2018: updates to sygus term database

### PR DESCRIPTION
**Pull request by ajreynol**
This is PR 13/18 from the sygus comp 2018 branch (https://github.com/ajreynol/CVC4/tree/sygusComp2018-2).

### CODE REVIEW

File: src/theory/quantifiers/sygus/term_database_sygus.h [hunk 1]
```diff
@@ -371,6 +394,27 @@ class TermDbSygus {
...
+   *
+   * We also may infer that operator is constructable. For example,
+   * - canConstructKind( B, ite, ... ) may return true, adding { B, B, B } to
+   * arg_types, noting that the term
+   *   (and (or (not b1) b2) (or b1 b3)) is equivalent to (ite b1 b2 b3)
+   */
+  bool canConstructKind(TypeNode tn,
+                        Kind k,
+                        std::vector<TypeNode>& argts,
+                        bool aggr = false);
```
Comment (HanielB) [inline src/theory/quantifiers/sygus/term_database_sygus.h]: document `aggr`

File: src/theory/quantifiers/sygus/term_database_sygus.cpp [hunk 2]
```diff
@@ -1228,6 +1317,92 @@ bool TermDbSygus::isSymbolicConsApp(Node n) const
...
+    }
+    return true;
+  }
+  if (!options::sygusSymBreakAgg())
+  {
+    return false;
+  }
+  if (sygusToBuiltinType(tn).isBoolean())
+  {
+    if (k == ITE)
```
Comment (HanielB) [inline src/theory/quantifiers/sygus/term_database_sygus.cpp]: join these `if`

Comment (ajreynol) [reply src/theory/quantifiers/sygus/term_database_sygus.cpp]: I prefer these separate.

**Comment by ajreynol** [comment]
I've addressed the review.

**Pull request merged**